routemaster/controllers/subscriber.rb

```diff
@@ -23,8 +23,7 @@ class Subscriber < Sinatra::Base
         end
 
         topics = data['topics'].map do |name|
-          Models::Topic.find(name) ||
-          Models::Topic.new(name: name, publisher: nil)
+          Models::Topic.find_or_create!(name: name)
```

Comment:
> Since this can raise the return code would be 500 and not 404, is that expected? I didn't see any changes to the relevant test.

As this is an optimisation-only ticket there shouldn't be any API changes.
The one exception this raises explicitly (hence the `!`) is `TopicClaimedError` — which is captured just below this comment.

> Also minor, as it's mostly a syntactic issue I have but is it okay to define these railsy methods in routemaster?

Well, why not? Hopefully it helped readability for you? Not sure I follow :)